dockers/README.md

```diff
@@ -0,0 +1,227 @@
+Building and maintaining these images can be challenging; this document provides a set of guidelines to assist in developing docker images for WARP.
+
+## <a name="goals"></a> Goals
+
+The following are some goals/guidelines we want to strive for when writing our Dockerfiles.
+
+### <a name="small"></a> Small images
+
+Building a smaller image offers advantages such as faster upload and download times along with reduced storage costs and minimized attack vector. Two of the easiest ways to minimize the size of your image is to use a small base image and to reduce the number of layers in your image.
+
+#### <a name="alpine"></a> Alpine base
+
+The easiest way to have a small image is to use an [Alpine](https://alpinelinux.org/) base image. Alpine linux, compared to Debian, RHEL etc., is designed specifically for security and resource efficiency, and lends itself perfectly to be used as a building block for Docker images.
+
+Along with being a small base, Alpine also has built in deletion of package index and provides [tini](https://github.com/krallin/tini) natively through APK.
+
+There are some instances where a Debian base image is unavoidable, specifically in the case where dependencies don't exists in APK. It is suggested that you only go to a Debian base as a last resort.
+
+
+##### :eyes: Example
+```dockerfile
+
+#OKAY, NOT GREAT - uses debian
+FROM python:debian
+
+RUN set eux; \
+        apt-get-update; \
+        apt-get install -y \
+            curl \
+            bash \
+    ; \
+# Must clean up cache manually with Debian
+    apt-get clean && rm -rf /var/lib/apt/list/*
+
+# GOOD - uses alpine
+FROM alpine:3.9
+
+RUN set eux; \
+        apk add --no-cache \
+            curl \
+            bash \
+```
+
+#### <a name="minimal-run"></a> Minimal RUN steps
+
+Having minimal `RUN`steps (ideally one) is another highly effective way to reduce the size of your image. Each instruction in a Dockerfile creates a [layer](https://docs.docker.com/storage/storagedriver/) and these layers are what add up to build the final image.
+When you use multple `RUN` steps it creates additional unnecessary layers and bloats your image.
+
+An alternative to having a single `RUN` step is to use [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/) which are effective when the application your are containerizing is just a statically linked binary. 
+Just to note, many of the images maintained in WARP require a handful of system-level dependencies and custom packages so multi-stages builds are typically not used.
+
+##### :eyes: Example
+```dockerfile
+
+# BAD - uses multiple RUN steps
+RUN set eux
+RUN apk add --no-cache curl bash wget
+RUN wget https://www.somezipfile.com/zip
+RUN unzip zip
+
+# GOOD - uses single RUN step
+RUN set eux; \
+        apk add --no-cache \
```

Comment:
Not sure if it is just my view, but it looks like the spacing might be a little weird here (?)